cis194/week4/stevemao/Homework.hs

```diff
@@ -1,6 +1,33 @@
 {-# OPTIONS_GHC -Wall #-}
 module Homework where
 
+fun1 :: [Integer] -> Integer
+fun1 [] = 1
+fun1 (x:xs)
```

Comment:
Can we get rid of the recursion calls here?

Reply:
- `otherwise = fun1 xs` means `if x is not even, I'll skip all the way to the next even element and apply fun1 on it`
- `even x = (x - 2) * fun1 xs` means `if x is even, I'll subtract x with 2 and multiply it by the next value`
So `func1 === "subtract all the even numbers by 2 and multiply the results together"`

Reply:
This is in the original Question I just copied it here. My solution is below: `fun1'`.

Reply:
> This is in the original Question I just copied it here. My solution is below: `fun1'`.

Ops.. my bad. Sorry...